src/chains/ethereum/ethereum/tests/api/eth/getProof.test.ts

```diff
@@ -0,0 +1,362 @@
+        assert.equal(
+          result.codeHash,
+          Data.toString(KECCAK256_NULL),
+          "Unexpected codeHash, expected keccak hash of zero bytes"
+        );
+
+        assert.equal(result.nonce, "0x1", "Unexpected nonce");
+        assert.equal(
+          result.storageHash,
+          emptyMerkleRoot,
+          "Unexpected storageHash, expected root of empty merkle trie"
+        );
+        assert.deepEqual(result.storageProof, [], "Unexpected storageProof");
+      });
+
+      it("gets the proof without specific storage indices", async () => {
+        const result = await provider.send("eth_getProof", [
+          contractAccount.address.toString(),
+          [],
+          "latest"
+        ]);
+
+        assert.equal(
+          result.address,
+          contractAccount.address.toString(),
+          "Unexpected address"
+        );
+        assert.equal(
+          result.balance,
+          contractAccount.balance.toString(),
+          "Unexpected balance"
+        );
+        assert.equal(
+          result.codeHash,
+          Data.toString(contractAccount.codeHash),
+          "Unexpected codeHash"
+        );
+        assert.equal(
+          result.nonce,
+          contractAccount.nonce.toString(),
+          "Unexpected nonce"
+        );
+        assert.equal(
+          result.storageHash,
+          Data.from(contractAccount.stateRoot).toString(),
+          "Unexpected storageHash"
+        );
+        assert.deepEqual(
+          result.accountProof,
+          expectedContractAccountProof,
+          "Unexpected accountProof"
+        );
+        assert.deepEqual(result.storageProof, [], "Unexpected storageProof");
+      });
+
+      it("gets the proof with specific storage index", async () => {
+        const storageSlotIndices = [Data.toString("0x0", 32), Data.toString("0x1", 32)];
+
+        const result = await provider.send("eth_getProof", [
+          contractAccount.address.toString(),
+          storageSlotIndices,
+          "latest"
+        ]);
+
+        assert.equal(
+          result.address,
+          contractAccount.address.toString(),
+          "Unexpected address"
+        );
+        assert.equal(
+          result.balance,
+          contractAccount.balance.toString(),
+          "Unexpected balance"
+        );
+        assert.equal(
+          result.codeHash,
+          Data.from(contractAccount.codeHash).toString(),
+          "Unexpected codeHash"
+        );
+        assert.equal(
+          result.nonce,
+          contractAccount.nonce.toString(),
+          "Unexpected nonce"
+        );
+        assert.equal(
+          result.storageHash,
+          Data.from(contractAccount.stateRoot).toString(),
+          "Unexpected storageHash"
+        );
+        assert.deepEqual(
+          result.accountProof,
+          expectedContractAccountProof,
+          "Unexpected accountProof"
+        );
+
+        const expectedStorageProof = [
```

Comment:
I was playing around with this on my own and couldn't get it to work. I am deploying a contract with the same code you're using for this contract. I'd get the transaction receipt to find the contract address, and call `eth_getProof` with:
 `[ contractAddress, ["0x0000000000000000000000000000000000000000000000000000000000000000", "0x0000000000000000000000000000000000000000000000000000000000000001"], "latest"]`. 

I don't get any error, but the storage proof is: `[{"key":"0x0000000000000000000000000000000000000000000000000000000000000000","proof":[],"value":"0x0"},{"key":"0x0000000000000000000000000000000000000000000000000000000000000001","proof":[],"value":"0x0"}]`

Any thoughts on why the value and proof is empty for my attempts?

Reply:
It looks like the create contract transaction is failing with `out of gas`. I passed `"gasLimit":"0x15f900"` and got the expected proof.

Reply:
Makes sense, thanks!